### ISSUE TITLE
Extremly slow linting when sqlalchemy involved

### ISSUE THREAD

**Issue by Zahlii** [PyCQA/pylint#8241]
### Bug description

I am currently running pylint on a (still relatively small, around 2000 LOC) Flask + SQLAlchemy project. However, applying pylint takes around 35 Minutes, even with similarity checks disabled.
 However, If I uninstall sqlalchemy, linting time goes down to a rough 4-5 seconds, which is much closer to what I would expect (obviously though, some import errors are found).
With sqlalchemy installed, but --disable=all, it still takes around 1-2 minutes to finish.

Note that i am perfectly aware of the -j option; however even if using all cores the time still only goes down to 11 minutes, which is not acceptable as a pre-commit hook.

I ran a cprofile run once WITH sqlalchemy and the config, and WITHOUT it installed. If needed, I can provide both of them; but I will post the main times here. It seems that with sqlalchemy, the astroid.infer functionality is invoked much more often causing this huge slowdown.

![without_sa](https://user-images.githubusercontent.com/218582/217608240-4960fb90-6ac2-4911-87a9-b8b1fb45465d.png)
![with_sa](https://user-images.githubusercontent.com/218582/217608262-ed5314e2-dc98-4aca-9f1a-843392492698.png)

### Configuration

```ini
Name: astroid
Version: 2.14.1
Summary: An abstract syntax tree for Python with inference support.
Home-page: 
Author: 
Author-email: Python Code Quality Authority <code-quality@python.org>
License: LGPL-2.1-or-later
Location: /Users/niklas.fruehauf/.pyenv/versions/3.9.13/envs/flow-venv-3.9/lib/python3.9/site-packages
Requires: lazy-object-proxy, typing-extensions, wrapt
Required-by: pylint
---
Name: pylint
Version: 2.16.1
Summary: python code static checker
Home-page: 
Author: 
Author-email: Python Code Quality Authority <code-quality@python.org>
License: GPL-2.0-or-later
Location: /Users/niklas.fruehauf/.pyenv/versions/3.9.13/envs/flow-venv-3.9/lib/python3.9/site-packages
Requires: astroid, dill, isort, mccabe, platformdirs, tomli, tomlkit, typing-extensions
Required-by:
```

### Command used

```shell
pylint -j 1 FOLDER/
```

### Pylint output

```shell
Normal output after 35+ Minutes
```

### Expected behavior

Normal output should happen after 4-5 seconds.

### Pylint version

```shell
pylint 2.16.1
astroid 2.14.1
Python 3.9.13 (main, Sep 21 2022, 15:17:17)
[Clang 13.0.0 (clang-1300.0.29.30)]
```

### OS / Environment

Mac OS

### Additional dependencies

sqlalchemy==2.0.0

**Comment by clavedeluna** [comment]
This issue has already been detected :)

Closing as a duplicate of https://github.com/PyCQA/pylint/issues/8150

**Comment by Pierre-Sassoulas** [comment]
> if using all cores the time still only goes down to 11 minutes, which is not acceptable as a pre-commit hook.

Outch. Yes, indeed :) Thank you for the c-profile, much appreciated!
